centreon/www/front_src/src/Resources/Filter/Filter.cypress.spec.tsx

```diff
@@ -386,169 +206,102 @@ before(() => {
-        result: [resourcesByHostType]
-      }
-    });
-
-    const endpointByServiceType = getListingEndpoint({
-      limit: 10,
-      resourceTypes: ['service'],
-      sort: {},
-      states: [],
-      statusTypes: [],
-      statuses: []
+      response
     });
+  });
 
-    cy.interceptAPIRequest({
-      alias: 'GetResourcesByServiceType',
-      method: Method.GET,
-      path: endpointByServiceType,
-      query: {
-        name: 'types',
-        value: '["service"]'
-      },
-      response: {
-        meta: { limit: 10, page: 1, search: {}, sort_by: {}, total: 1 },
-        result: [resourcesByServiceType]
-      }
-    });
+  cy.fixture('resources/filter/resourcesByServiceType.json').then(
+    (response) => {
+      cy.interceptAPIRequest({
+        alias: 'GetResourcesByServiceType',
+        method: Method.GET,
+        path: `${resourcesEndpoint}**`,
+        query: {
+          name: 'types',
+          value: '["service"]'
+        },
+        response
+      });
+    }
+  );
 
+  cy.fixture('resources/filter/pollers.json').then((response) => {
     cy.interceptAPIRequest({
       alias: 'pollersRequest',
       method: Method.GET,
       path: '**/monitoring/servers?*',
-      response: {
-        meta: { limit: 10, page: 1, search: {}, sort_by: {}, total: 1 },
-        result: [pollersData]
-      }
+      response
     });
+  });
 
+  cy.fixture('resources/filter/hostGroups.json').then((response) => {
     cy.interceptAPIRequest({
-      alias: 'hostgroupsRequest',
+      alias: 'hostGroupsRequest',
       method: Method.GET,
       path: '**/hostgroups?*',
-      response: {
-        meta: { limit: 10, page: 1, search: {}, sort_by: {}, total: 1 },
-        result: [linuxServersHostGroup]
-      }
+      response
     });
+  });
 
+  cy.fixture('resources/filter/webAccessServiceGroup.json').then((response) => {
     cy.interceptAPIRequest({
       alias: 'serviceGroupsRequest',
       method: Method.GET,
       path: '**/servicegroups?*',
-      response: {
-        meta: { limit: 10, page: 1, search: {}, sort_by: {}, total: 1 },
-        result: [webAccessServiceGroup]
-      }
+      response
     });
+  });
+
+  cy.fixture('resources/filter/hostCategory.json').then((response) => {
     cy.interceptAPIRequest({
       alias: 'hostCategoryRequest',
       method: Method.GET,
       path: '**/monitoring/hosts/categories?*',
-      response: {
-        meta: { limit: 10, page: 1, search: {}, sort_by: {}, total: 1 },
-        result: [hostCategoryData]
-      }
+      response
     });
+  });
 
+  cy.fixture('resources/filter/hostSeverity.json').then((response) => {
     cy.interceptAPIRequest({
       alias: 'hostSeverityRequest',
       method: Method.GET,
       path: '**/monitoring/severities/host?*',
-      response: {
-        meta: { limit: 10, page: 1, search: {}, sort_by: {}, total: 1 },
-        result: [hostSeverityData]
-      }
+      response
     });
+  });
```

Comment:
humm

Reply:
each request needs to be initialized with the function interceptAPIRequest (which is a generic function ) , and each request have his own data that can be loaded via cy.fixture , so i don t see an extra optimisation for this , otherwise i want a suggestion holding this conditions

Reply:
ok

Reply:
i pushed a fix in commit 321268b